### PR TITLE
Add slsutil.renderer execution function

### PR DESCRIPTION
**Pull request by whiteinge**
### What does this PR do?

Add an `slsutil.renderer` function. See the docstring for usage.
### Tests written?
- [ ] Yes
- [x] No

### CODE REVIEW

File: salt/modules/slsutil.py [hunk 1]
```diff
@@ -33,3 +36,87 @@
...
+    One possible use is to allow writing "map files", as are commonly seen in
+    Salt formulas, but without tying the renderer of the map file to the
+    renderer used in the other sls files. In other words, a map file could use
+    the Python renderer and still be included and used by an sls file that uses
+    the default 'jinja|yaml' renderer.
+
+    For example, the two following map files produce identical results but one
+    is written using the normal 'jinja|yaml' and the other is using 'py':
+
+    .. code-block:: jinja
```
Comment (bechtoldt) [inline salt/modules/slsutil.py]: you have ``salt['mod.fun']`` and ``salt.mod.fun`` here, wouldn't it make sense to use only one of them?

**Comment by bechtoldt** [comment]
wouldn't it make sense to combine this with https://github.com/saltstack/salt/blob/develop/salt/modules/defaults.py ?

File: salt/modules/slsutil.py (hunk 1 above)
Comment (whiteinge) [inline salt/modules/slsutil.py]: Thanks. Copy-and-paste overlook.

**Comment by whiteinge** [comment]
Possibly. What'd you have in mind?

**Comment by rallytime** [comment]
@whiteinge Can you grab the [lint error](https://jenkins.saltstack.com/job/salt-pr-lint-n/14194/violations/file/salt/modules/slsutil.py/)? The other failures are not related (unless this is changing based on the comments above).

**Comment by whiteinge** [comment]
Go Go Jenkins!

**Comment by whiteinge** [comment]
@rallytime sorry for the slow reply. Updated.

**Pull request merged**